src/core/symbology/qgslinesymbollayer.cpp

```diff
@@ -1825,10 +1861,40 @@ void QgsTemplatedLineSymbolLayerBase::renderPolylineVertex( const QPolygonF &poi
     context.setOriginalValueVariable( mOffsetAlongLine );
     offsetAlongLine = mDataDefinedProperties.valueAsDouble( QgsSymbolLayer::PropertyOffsetAlongLine, context.renderContext().expressionContext(), mOffsetAlongLine );
   }
+
+  // only calculated if we need it!!
+  double totalLength = -1;
   if ( !qgsDoubleNear( offsetAlongLine, 0.0 ) )
   {
+    totalLength = QgsSymbolLayerUtils::polylineLength( points );
```

Comment:
if there is a cost to this calculation, it could be a bit more fine tuned and calculated strictly where needed? or it's not worth it?

Reply:
@3nids done in https://github.com/qgis/QGIS/pull/46494